### DISCUSSION TITLE
Client certificates via ansible modules

### DISCUSSION THREAD

**Discussion by wenners** [netbox-community/netbox#5427] [Q&A] [answered]
Hi,

I’m struggling with ansible and netbox in regard to the usage of client certificates.
Reading the documentation of pynetbox I found nice examples (https://pynetbox.readthedocs.io/en/latest/advanced.html#custom-sessions) how to use an existing python requests session where I can add headers and also use client certificates for authentication.
After digging and searching all the docs for the ansible collection (https://netbox-ansible-collection.readthedocs.io/en/latest/) I could not find out how to replicate this behavior in ansible. 

Any pointers are really appreciated.

Thanks
Sebastian

**Comment by nahun** [comment, marked as answer]
There isn't a way to replicate that with the Ansible modules. That would take an enhancement in the ansible_modules repo. You can send an issue requesting that if you want. I don't know if it will happen, but you can always ask.
